app/controllers/api/harvests_controller.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module Api
+  # Initiate a harvest via the API (e.g. from Airflow)
+  class HarvestsController < ApiController
+    delegate :t, to: :I18n
+
+    def create
+      begin
+        upload = NdjsonUpload.new(params.require(:url))
+        return invalid_file unless upload.valid?
+      rescue ActionController::ParameterMissing
+        return no_url
+      rescue Errno::ENOENT
```

Comment:
Can we move a check for `File.exists?` into `upload.valid?` Then it doesn't have to raise an error in this case?

Reply:
How would we handle the two error cases separately? I think we still want to preserve the distinction between 422 (filepath OK, bad contents) and 400 (filepath not OK)

Reply:
Have an `error` method on the service object. Once you determine that it's not valid, then you can expect the error method to return a value which can be decoded into a response.

Reply:
I don't think I understand this, but happy to pair if you're willing to explain it